pkg/storage/gcs/gcs.go

```diff
@@ -0,0 +1,128 @@
+package gcs
+
+import (
+	"reflect"
+
+	coreapi "k8s.io/api/core/v1"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/clusterconfig"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+)
+
+type driver struct {
+	Config  *imageregistryv1.ImageRegistryConfigStorageGCS
+	Listers *regopclient.Listers
+}
+
+func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageGCS, listers *regopclient.Listers) *driver {
+	return &driver{
+		Config:  c,
+		Listers: listers,
+	}
+}
+
+func (d *driver) Secrets() (map[string]string, error) {
+	cfg, err := clusterconfig.GetGCSConfig(d.Listers)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]string{
+		"STORAGE_GCS_KEYFILE": cfg.Storage.GCS.KeyfileData,
+	}, nil
+}
+
+func (d *driver) ConfigEnv() (envs []coreapi.EnvVar, err error) {
+	envs = append(envs,
+		coreapi.EnvVar{Name: "REGISTRY_STORAGE", Value: "gcs"},
+		coreapi.EnvVar{Name: "REGISTRY_STORAGE_GCS_BUCKET", Value: d.Config.Bucket},
+		coreapi.EnvVar{Name: "REGISTRY_STORAGE_GCS_KEYFILE", Value: "/gcs/keyfile"},
+	)
+	return
+}
+
+func (d *driver) Volumes() ([]coreapi.Volume, []coreapi.VolumeMount, error) {
+	vol := coreapi.Volume{
+		Name: "registry-storage-keyfile",
+		VolumeSource: coreapi.VolumeSource{
+			Projected: &coreapi.ProjectedVolumeSource{
```

Comment:
side note - `ProjectedVolumeSource` is new to me (and I like it!)

Looks like it's GA in kube 1.13, so no concerns here w.r.t upgrades.